hail/python/hail/backend/service_backend.py

```diff
@@ -8,14 +10,43 @@
 
 from hailtop.config import get_deploy_config, get_user_config
 from hailtop.auth import service_auth_headers
-from hailtop.utils import (retry_response_returning_functions,
-                           external_requests_client_session)
+from hailtop.utils import async_to_blocking
 from hail.ir.renderer import CSERenderer
 
 from .backend import Backend
 from ..hail_logging import PythonOnlyLogger
 
 
+class ServiceSocket:
```

Comment:
As written, this assumes you can hold a web socket open for a long time. In general, Hail services will get restarted. I think `async_request` needs to be re-written to reconnect and retry the whole request if it encounters a `CLOSED`.

Longer term, we probably want operation UUIDs so that if the socket hangs up but the service is still alive, we can reconnect and wait on the same operation rather than starting a second one.

Reply:
Sounds good. I'll fix it to retry on `CLOSED` as soon as I figure out why the socket is timing out after 60 seconds even without the heartbeat.